In [5]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

import scipy.stats as stats

# Input the data

In [7]:
oscheme = np.array([57,103,59,75,84,73,35,110,44,82,67,64,78,53,41,39,80,87,73,65,28,62,49,84,63,77,67,101,91,50]) #Old Scheme data
nscheme = np.array([62,122,54,82,84,86,32,104,38,107,84,85,99,39,34,58,73,53,66,78,41,71,38,95,81,58,75,94,100,68]) #New Scheme data

1.	Find the mean of old scheme and new scheme column. 

In [9]:
omean= np.mean(oscheme)
nmean= np.mean(nscheme)

print('The mean of old scheme column is %3.2f' %omean)
print('The mean of new scheme column is %3.2f' %nmean)

The mean of old scheme column is 68.03
The mean of new scheme column is 72.03


2.	Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? 

Answer in Step 4 of the Hypothesis below

3.	What conclusion does the test (p-value) lead to? 

Answer in Step 5 of the Hypothesis below

### Step 1: Define null and alternative hypotheses

In testing whether output from new scheme is significantly different from old scheme, the null hypothesis states that the mean monthly output, $\mu{Old Scheme}$ equals $\mu{New Scheme}$. The alternative hypothesis states that difference in output is more than 0, $\mu{Old Scheme}$ $\neq$ $\mu{New Scheme}$

* $H_0$: $\mu{Old Scheme}$ - $\mu{New Scheme}$ =  0
* $H_A$: $\mu{Old Scheme}$ - $\mu{New Scheme}$ $\neq$  0

### Step 2: Decide the significance level

Here we select $\alpha$ = 0.05 and population standard deviation is not known.

### Step 3: Identify the test statistic

* Sample sizes for both samples are  same.
* We have two paired samples and we do not know the population standard deviation.
* We use the t distribution and the $t_STAT$ test statistic for two sample paired test.

### Step 4: Calculate the p - value and test statistic

In [49]:
t_statistic, p_value  =  stats.ttest_rel(oscheme, nscheme)
print('P Value %1.3f' % p_value)  

P Value 0.131


In [24]:
p_value > 0.05 #p critical value is 5%, i.e., 0.05

True

### Step 5:  Decide to reject or accept null hypothesis

The p value is 0.131 which is more than 5% level of significance

So, the statistical decision is fail to reject the null hypothesis at 5% level of significance.

### So there is no sufficient evidence that the montly output with old scheme and new scheme are not equal. So we will say that outputs are similar with old scheme and new scheme

4.	Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:

If we increase the avg output in new scheme by 5k, we will get the new array and then the values will be

In [54]:
nschemeincr = nscheme +5

t_statistic, p_value  =  stats.ttest_rel(oscheme, nschemeincr)
print('P Value for the difference %1.3f' % p_value) 

P Value for the difference 0.002


a) The probability of a type 1 error? 

As the question suggested that after adding 5k, we get the actual value to make break even, for which we calculated the pvalue of 0.002, our alpha value is 0.05

which means for all the value above 0.002 the null hypothesis should be true

The type I error is when the actual null hypothesis is true but we rejected the null hypothsis

we are rejecting the null hypothesis after 5% level of significance when it should be 0.2%

so the probability of making this error is :

In [53]:
alpha = 0.05

probtype1 = (alpha-p_value)/ alpha

print('The probability of making type 1 error is %1.3f' %probtype1)

The probability of making type 1 error is 0.970


b) What is the p- value of the hypothesis test if we test for a difference of $5000?

The p value of the hypothesis testing for a difference of 5k is 0.002 which is less than our level of significance of 0.05 (5%).

So we will reject the null hypothesis and state that there is a significance difference in new and old scheme output

### The output with old scheme is significantly lower than the output with new scheme

The End